tools/autograd/gen_variable_type.py

```diff
@@ -747,17 +746,25 @@ def unpack_args(env, declaration):
 
             dynamic_type = arg['dynamic_type']
             is_nullable = arg.get('is_nullable', False)
+            extra_args = []
             ref = (not is_nullable) and dynamic_type not in ['TensorList', 'SparseTensor']
-            suffix = get_suffix(dynamic_type, is_nullable)
+            suffix = ''
             if dynamic_type == 'TensorList' and declaration['name'] == 'index':
                 # TODO: specify this in Declarations.yaml somehow
                 suffix = '_idxs'
+            elif dynamic_type in SCALAR_TYPE_MAP:
+                suffix = '_type'
+                extra_args.append(SCALAR_TYPE_MAP[dynamic_type])
```

Comment:
if this isn't generic, just set it (so you can look on one line and know what it is), e.g.
`type_arg =[SCALAR_TYPE_MAP[dynamic_type]`